docs/api/other-api/adapter.md

```diff
@@ -0,0 +1,191 @@
+  console.log(
+    `Server is running on http://localhost:${PORT}`,
+  );
+});
+```
+
+### Migrating from the React Router App Server
+
+If you started an app with the [React Router App Server][rr-serve] but find that you want to take control over the Express server and customize it, it should be fairly straightforward to migrate way from `@react-router/serve`.
+
+You can refer to the [Express template][express-template] as a reference, gut here's the main changes you should have to make:
+
+**1. Update deps**
+
+```sh
+npm uninstall @react-router/serve
+npm install @react-router/express compression express morgan cross-env
+npm install --save-dev @types/express @types/express-serve-static-core @types/morgan
+```
+
+**2. Add a server**
+
+Create your React Router express server in `server/app.ts`:
+
+```ts filename=server/app.ts
+import "react-router";
+import { createRequestHandler } from "@react-router/express";
+import express from "express";
+
+export const app = express();
+
+app.use(
+  createRequestHandler({
+    build: () =>
+      import("virtual:react-router/server-build"),
+  }),
+);
+```
+
+Copy the [`server.js`][express-template-server-js] into your app. This is the boilerplate setup we recommend to allow the same server code to run both the development and production builds of your app. Two separate files are used here so that the main Express server code can be written in TypeScript (`server/app.ts`) and compiled into your server build by React Router, and then executed via `node server.js`.
+
+**3. Update `vite.config.ts` to compile the server**
+
+```tsx filename=vite.config.ts lines=[6-10]
+import { reactRouter } from "@react-router/dev/vite";
+import { defineConfig } from "vite";
+import tsconfigPaths from "vite-tsconfig-paths";
+
+export default defineConfig(({ isSsrBuild }) => ({
+  build: {
+    rollupOptions: isSsrBuild
+      ? { input: "./server/app.ts" }
+      : undefined,
+  },
+  plugins: [reactRouter(), tsconfigPaths()],
+}));
+```
+
+**4. Update `package.json` scripts**
+
+Update the `dev` and `start` scripts to use your new Express server:
+
+```json filename=package.json
+{
+  // ...
+  "scripts": {
+    "dev": "cross-env NODE_ENV=development node server.js",
+    "start": "node server.js"
+    // ...
+  }
+  // ...
+}
+```
+
+## `@react-router/architect`
+
+Here's an example with Architect (AWS):
+
+```ts
+const {
+  createRequestHandler,
+} = require("@react-router/architect");
+exports.handler = createRequestHandler({
+  build: require("./build"),
+});
+```
+
+## `@react-router/cloudflare`
+
+Here's an example with the simplified Cloudflare Workers API:
+
+```ts
+import { createEventHandler } from "@react-router/cloudflare-workers";
+
+import * as build from "../build";
+
+addEventListener("fetch", createEventHandler({ build }));
+```
+
+<!-- TODO: We used to have a Community Adapters section here, but unsure which of those are RR friendly so we should check that before re-including? -->
```

Comment:
@brookslybrand Curious what your thoughts are here and if we want to try to get something like https://remix.run/docs/en/main/other-api/adapter#community-adapters added back